web/docs/auth/username-and-pass.md

```diff
@@ -482,19 +484,29 @@ import {
   ensureValidPassword,
   ensureValidUsername,
 } from '@wasp/auth/validation.js'
+import {
+  createProviderId,
+  sanitizeAndSerializeProviderData,
+  createUser,
+} from '@wasp/auth/utils.js'
 
 export const signup = async (args, { entities: { User } }) => {
   ensureValidUsername(args)
   ensurePasswordIsPresent(args)
   ensureValidPassword(args)
 
   try {
-    await User.create({
-      data: {
-        username: args.username,
-        password: args.password, // Password is hashed automatically by Wasp
-      },
+    const providerId = createProviderId('username', args.username)
+    const providerData = await sanitizeAndSerializeProviderData({
+      hashedPassword: args.password,
     })
+
+    await createUser(
```

Comment:
You'll notice I've opted to use our helper functions since it's easier to implement the signup flow vs. using plain Prisma.

Reply:
Sure, well that makes sense!

Here also I think we could benefit from higher-level functions, and of course hooks.

Reply:
Yep, I agree, here's an issue: https://github.com/wasp-lang/wasp/issues/1631